pkg/cloud/prepare/aws.go

```diff
@@ -19,57 +19,40 @@ limitations under the License.
 package prepare
 
 import (
-	"github.com/spf13/cobra"
 	"github.com/submariner-io/cloud-prepare/pkg/api"
-	"github.com/submariner-io/submariner-operator/internal/exit"
-	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/cloud/aws"
+	"github.com/submariner-io/submariner-operator/pkg/cloud"
+	"github.com/submariner-io/submariner-operator/pkg/cloud/aws"
+	"k8s.io/client-go/rest"
 )
 
-// NewCommand returns a new cobra.Command used to prepare a cloud infrastructure.
-func newAWSPrepareCommand() *cobra.Command {
-	cmd := &cobra.Command{
-		Use:   "aws",
-		Short: "Prepare an OpenShift AWS cloud",
-		Long:  "This command prepares an OpenShift installer-provisioned infrastructure (IPI) on AWS cloud for Submariner installation.",
-		Run:   prepareAws,
-	}
-
-	aws.AddAWSFlags(cmd)
-	cmd.Flags().StringVar(&awsGWInstanceType, "gateway-instance", "c5d.large", "Type of gateways instance machine")
-	cmd.Flags().IntVar(&gateways, "gateways", DefaultNumGateways,
-		"Number of dedicated gateways to deploy (Set to `0` when using --load-balancer mode)")
-
-	return cmd
-}
-
-func prepareAws(cmd *cobra.Command, args []string) {
+func Aws(port cloud.Port, instance cloud.Instance, config *rest.Config, reporter api.Reporter) error {
```

Comment:
Pass client interfaces (or `client.Producer`) instead of `rest.Config`.

```suggestion
func AWS(port cloud.Port, instance cloud.Instance, config *rest.Config, reporter api.Reporter) error {
```